arlas-tests/src/test/java/io/arlas/server/rest/explore/AbstractFilteredTest.java

```diff
@@ -231,507 +231,516 @@ public void testRangeFilter() throws Exception {
         handleMatchingGeometryFilter(header(request.filter), 17, everyItem(endsWith("170")));
 
         /** counter clock-wise WKT */
-        request.filter.notpwithin = Arrays.asList(new MultiValueFilter<>("POLYGON((-2 -2, 2 -2, 2 2,-2 2, -2 -2))"));
+        request.filter.f = Arrays.asList(new MultiValueFilter<>(new Expression("geo_params.centroid", OperatorEnum.notwithin, "POLYGON((-2 -2, 2 -2, 2 2,-2 2, -2 -2))")));
         handleMatchingGeometryFilter(post(request), 561, everyItem(notNullValue()));
-        handleMatchingGeometryFilter(get("notpwithin", request.filter.notpwithin.get(0).get(0)), 561, everyItem(notNullValue()));
+        handleMatchingGeometryFilter(get("f", request.filter.f.get(0).get(0).toString()), 561, everyItem(notNullValue()));
         handleMatchingGeometryFilter(header(request.filter), 561, everyItem(notNullValue()));
 
         /** west < east bbox*/
-        request.filter.notpwithin = Arrays.asList(new MultiValueFilter<>("-175,-85,175,85"));
+        request.filter.f = Arrays.asList(new MultiValueFilter<>(new Expression("geo_params.centroid", OperatorEnum.notwithin, "-175,-85,175,85")));
         handleNotMatchingNotPwithinFilter(post(request));
-        handleNotMatchingNotPwithinFilter(get("notpwithin", request.filter.notpwithin.get(0).get(0)));
+        handleNotMatchingNotPwithinFilter(get("f", request.filter.f.get(0).get(0).toString()));
         handleNotMatchingNotPwithinFilter(header(request.filter));
 
         /** clock-wise WKT */
-        request.filter.notpwithin = Arrays.asList(new MultiValueFilter<>("POLYGON((-175 -85, -175 85, 175 85, 175 -85, -175 -85))"));
+        request.filter.f = Arrays.asList(new MultiValueFilter<>(new Expression("geo_params.centroid", OperatorEnum.notwithin, "POLYGON((-175 -85, -175 85, 175 85, 175 -85, -175 -85))")));
         handleNotMatchingNotPwithinFilter(post(request));
-        handleNotMatchingNotPwithinFilter(get("notpwithin", request.filter.notpwithin.get(0).get(0)));
+        handleNotMatchingNotPwithinFilter(get("f", request.filter.f.get(0).get(0).toString()));
         handleNotMatchingNotPwithinFilter(header(request.filter));
 
         /** counter clock-wise WKT */
-        request.filter.notpwithin = Arrays.asList(new MultiValueFilter<>(Arrays.asList("POLYGON((160 -2, 160 2, -160 2, -160 -2, 160 -2))")));
+        request.filter.f = Arrays.asList(new MultiValueFilter<>(new Expression("geo_params.centroid", OperatorEnum.notwithin, "POLYGON((160 -2, 160 2, -160 2, -160 -2, 160 -2))")));
         handleMatchingGeometryFilter(post(request), 592, everyItem(notNullValue()));
-        handleMatchingGeometryFilter(get("notpwithin", request.filter.notpwithin.get(0).get(0)), 592, notNullValue());
+        handleMatchingGeometryFilter(get("f", request.filter.f.get(0).get(0).toString()), 592, notNullValue());
         handleMatchingGeometryFilter(header(request.filter), 592, everyItem(notNullValue()));
 
```

Comment:
It will be good to add a test with `notwithin` querying with a `MULTIPOLYGON` (ref `andQueryBuilder` comment)

Reply:
Done